cmake/KokkosConfigCommon.cmake.in

```diff
@@ -226,7 +226,7 @@ FUNCTION(kokkos_compilation)
     SET(_Kokkos_ENABLE_CUDA @Kokkos_ENABLE_CUDA@)
 
     # search relative first and then absolute
-    SET(_HINTS "${CMAKE_CURRENT_LIST_DIR}/../.." "@CMAKE_INSTALL_PREFIX@")
+    SET(_HINTS "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/../../.." "@CMAKE_INSTALL_PREFIX@")
```

Comment:
Note that `CMAKE_CURRENT_FUNCTION_LIST_DIR` appeared in CMake 3.17 ([doc](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_FUNCTION_LIST_DIR.html)), which is higher than the current minimum version of CMake required by Kokkos is 3.16.

Reply:
This is problematic, see the below reproducer. Perhaps bumping to 3.17 might be best other options seems hackish.

Reply:
From #3679, it's hard to say why 3.16 was chosen over 3.17. In addition, Trilinos requires 3.23 now.

Reply:
The reason maybe because it was the cmake shipped by ubuntu 20.04, in the cmake project that I participate I try to stay compatible with the previous ubuntu LTS (22.04 now).